src/main/java/htsjdk/samtools/util/AbstractProgressLogger.java

```diff
@@ -80,14 +92,22 @@ public synchronized boolean log() {
         }
     }
 
-    @Override
-    public synchronized boolean record(final String chrom, final int pos) {
-        this.lastChrom = chrom;
-        this.lastPos = pos;
-        if (this.lastStartTime == -1) {
-            this.lastStartTime = System.currentTimeMillis();
+    protected synchronized boolean record(final String chrom, final int pos, final String rname) {
+        lastChrom = chrom;
+        if (pos >= lastPos) {
```

Comment:
only increase counter when chrom name is unchanged, and do not decrease counter.